src/main/java/ch/heigvd/statique/utils/Server.java

```diff
@@ -0,0 +1,97 @@
+package ch.heigvd.statique.utils;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpServer;
+import org.apache.commons.io.FileUtils;
+
+import java.io.*;
+import java.net.InetSocketAddress;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Path;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.logging.Logger;
+
+public class Server {
+    private final static Logger LOG = Logger.getLogger(Server.class.getName());
+    private final Path site;
+    private final int port;
+
+    /**
+     * Server constructor
+     *
+     * @param site path to site
+     * @param port server port number
+     */
+    public Server(Path site, int port) {
+        this.site = site;
+        this.port = port;
+    }
+
+    /**
+     * Start the server
+     *
+     * @throws IOException Couldn't start the server
+     */
+    public void start() throws IOException {
+        HttpServer server = HttpServer.create(new InetSocketAddress(port), 1);
+        server.createContext("/", new MyHttpHandler());
+        ThreadPoolExecutor threadPoolExecutor = (ThreadPoolExecutor) Executors.newFixedThreadPool(10);
+        server.setExecutor(threadPoolExecutor);
+        server.start();
+        LOG.info(" Server started on port 1234");
+    }
+
+    class MyHttpHandler implements HttpHandler {
+
+        /**
+         * Handle http request
+         * @param t http exchange
+         * @throws IOException Couldn't respond to the client request
+         */
+        public void handle(HttpExchange t) throws IOException {
+            // Modified code from : http://www.microhowto.info/howto/serve_web_pages_using_an_embedded_http_server_in_java.html
+
+            // Gets request
+            URI uri = t.getRequestURI();
+            File file = new File(site + uri.getPath()).getCanonicalFile();
+
+            LOG.info("File path : " + file.getPath());
+
+            if(uri.toString().contains("/../")){
```

Comment:
This will allow access with `/..%2findex.md` (`%2f` correspond to `/`)

Use this to handle instead

```suggestion
            if(!file.getPath().startsWith(site.toString())) {
```